### PR TITLE
docs: add auto-generated reference doc for garden.yml configuration

### PR DESCRIPTION
**Pull request by edvald**

### CODE REVIEW

File: docs/reference/config.md [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+        copy: 
+          - # POSIX-style path or filename of the directory or file(s) to copy to the target.
+            # 
+            # Required.
+            source: 
+            
+            # POSIX-style path or filename to copy the directory or file(s) to (defaults to same
+            # as source path).
+            # 
+            # Optional.
```
Comment (Deepakkothandan) [inline docs/reference/config.md]: @edvald if target is optional, what is the default value ?

**Comment by edvald** [comment]
Arrrgh for some reason my local setup makes a different `package-lock.json` than the CI. I've just about had enough of these issues with `npm`.

**Comment by Deepakkothandan** [comment]
@edvald switch to yarn maybe ? 😄

**Comment by edvald** [comment]
Yeah I've been pondering that. Although I remember having a fair amount of issues with yarn stability as well...

File: docs/reference/config.md [hunk 2]
```diff
@@ -0,0 +1,582 @@
+## garden.yml reference
+
+Below is the full schema for the `garden.yml` configuration files. For a introductory guide
```
Comment (eysi09) [inline docs/reference/config.md]: "For _an_ introdcutory..."

Comment (eysi09) [reply docs/reference/config.md]: Actually: "For _an_ introductory guide_,_ please look _at_ our [configuration guide](../guides/configuration.md). (Note the command after guide).

File: docs/reference/config.md [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+please look our [configuration guide](../guides/configuration.md).
+
+Note that individual module types (e.g. container) add additional configuration keys. The built-in module types
```
Comment (eysi09) [inline docs/reference/config.md]: Maybe reads a bit clearer with: "Note that individual module types, e.g. container. add additional configuration keys."

Comment (eysi09) [reply docs/reference/config.md]: ...these comments actually apply to the `config.hbs` file, not the generated file.

**Comment by edvald** [comment]
Resolved the issues you pointed out @eysi09

**Comment by eysi09** [comment]
Just one more comment: would it make sense to split the `garden.yml` code block into `project` and `module` parts? So there would be two markdown code blocks, one for each directive respectively.

**Pull request merged**